esp-hal/src/uart.rs

```diff
@@ -739,7 +787,19 @@ where
     /// Change the configuration.
     ///
     /// Note that this also changes the configuration of the TX half.
-    // FIXME: when https://github.com/esp-rs/esp-hal/issues/2839 is resolved, add an appropriate `# Error` entry.
+    ///
+    /// # Errors
+    ///
+    /// [`ConfigError::UnsupportedFifoThreshold`] will be returned if the RX
+    /// FIFO threshold passed in `Config` exceeds the maximum value (
```

Comment:
Similar to my other comment, I'd avoid writing the conditions in long sentences when an error variant has more than one return condition, I think bullet pointing it should be clearer.